pkg/operation/botanist/controlplane/resourcemanager/resource_manager.go

```diff
@@ -0,0 +1,553 @@
+	ClusterRoleName            *string
+	DefaultLabels              *map[string]string
+	HealthSyncPeriod           *string
+	KubeConfig                 *component.Secret
+	LeaseDuration              *string
+	MaxConcurrentHealthWorkers *int32
+	RenewDeadline              *string
+	ResourceClass              *string
+	RetryPeriod                *string
+	SyncPeriod                 *string
+	TargetDisableCache         *bool
+	WatchedNamespace           *string
+}
+
+func (r *ResourceManager) Deploy(ctx context.Context) error {
+	if err := r.ensureServiceAccount(ctx); err != nil {
+		return err
+	}
+
+	if err := r.ensureRBAC(ctx); err != nil {
+		return err
+	}
+
+	if err := r.ensureService(ctx); err != nil {
+		return err
+	}
+
+	if err := r.ensureDeployment(ctx); err != nil {
+		return err
+	}
+
+	return r.ensureVPA(ctx)
+}
+
+func (r *ResourceManager) Destroy(ctx context.Context) error {
+	objectsToDelete := []client.Object{
+		r.emptyVPA(),
+		r.emptyDeployment(),
+		r.emptyService(),
+		r.emptyServiceAccount(),
+	}
+	role, err := r.emptyRoleInWatchedNamespace()
+	if err == nil {
+		objectsToDelete = append(objectsToDelete, role)
+	}
+	rb, err := r.emptyRoleBindingInWatchedNamespace()
+	if err == nil {
+		objectsToDelete = append(objectsToDelete, rb)
+	}
+	cr, err := r.emptyClusterRole()
+	if err == nil {
+		objectsToDelete = append(objectsToDelete, cr)
+	}
+	crb, err := r.emptyClusterRoleBinding()
+	if err == nil {
+		objectsToDelete = append(objectsToDelete, crb)
+	}
+
+	return kutil.DeleteObjects(
+		ctx,
+		r.client,
+		objectsToDelete...,
+	)
+}
+
+func (r *ResourceManager) ensureRBAC(ctx context.Context) error {
+	deployResourcesIntoAnotherCluster := r.config.KubeConfig != nil
+	if deployResourcesIntoAnotherCluster {
+		if r.config.WatchedNamespace == nil {
+			if err := r.ensureClusterRole(ctx, allowManagedResources); err != nil {
+				return err
+			}
+			if err := r.ensureClusterRoleBinding(ctx); err != nil {
+				return err
+			}
+		} else {
+			if err := r.ensureRoleInWatchedNamespace(ctx, allowManagedResources); err != nil {
+				return err
+			}
+			if err := r.ensureRoleBinding(ctx); err != nil {
+				return err
+			}
+		}
+	} else {
+		if err := r.ensureClusterRole(ctx, allowAll); err != nil {
+			return err
+		}
+		if err := r.ensureClusterRoleBinding(ctx); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *ResourceManager) ensureClusterRole(ctx context.Context, policies []rbacv1.PolicyRule) error {
+	clusterRole, err := r.emptyClusterRole()
+	if err != nil {
+		return err
+	}
+	_, err = controllerutil.CreateOrUpdate(ctx, r.client, clusterRole, func() error {
+		clusterRole.Labels = *r.config.DefaultLabels
+		clusterRole.Rules = policies
+		return nil
+	})
+	return err
+}
+
+func (r *ResourceManager) emptyClusterRole() (*rbacv1.ClusterRole, error) {
+	if r.config.ClusterRoleName == nil {
+		return nil, errors.New("creating Cluster Role failed - no name defined for the Cluster Role")
+	}
```

Comment:
Can you remove the `ClusterRoleName` from the config and rather compute it here?

Reply:
This would mean I would have to give it the name:
`gardener-resource-manager-seed` which I want to avoid as it ties the implementation to the current configuration of the central grm deployed in seeds.

Reply:
Yes, I know and understand, I tried the same in other packages, but it didn't always work, and in the end, this whole package is part of g/g and you already use such terms (like "seed" or "shoot") in other places, e.g. https://github.com/gardener/gardener/pull/3445/files#diff-a66ff079fa067226cf9c172079a286e54f39e2a5ec6ad863136488759de35274R377.
Hence, it doesn't make it worse but rather easier if this `ClusterRoleName` disappears from the configuration. It's anyways somewhat hard to come up with a meaningful value here as a "user" of this package because it's an implementation detail (I would have to look up the code to check where this setting is used to come up with a "good" value).

Reply:
I agree. Also made the tests easier. Thanks for your feedback!